src/Data/Avro/Deriving.hs

```diff
@@ -83,6 +102,32 @@ mkPrefixedFieldName :: TypeName -> Field -> T.Text
 mkPrefixedFieldName (TN dn) fld = sanitiseName $
   updateFirst T.toLower dn <> updateFirst T.toUpper (fldName fld)
 
+
+-- | Marks any field as non-strict in the generated data types.
+mkLazyField :: TypeName -> Field -> (FieldStrictness, FieldUnpackedness)
+mkLazyField _ _ =
+  (LazyField, NonUnpackedField)
+
+
+-- | Make a field strict and unpacked if it has a primitive representation.
+-- Primitive types are types which GHC has either a static or an unlifted
+-- representation: `()`, `Boolean`, `Int32`, `Int64`, `Float`, `Double`.
+mkStrictPrimitiveField :: TypeName -> Field -> (FieldStrictness, FieldUnpackedness)
+mkStrictPrimitiveField _ field =
```

Comment:
@AlexeyRaga please don't merge yet. I think this might cause warnings in GHC: Booleans have static representation but are sum types. These can't be unpacked. Let me fix it quickly.

Reply:
Okay, done.